Sources/TensorFlow/Layers/Normalization.swift

```diff
@@ -224,3 +244,166 @@ public struct LayerNorm<Scalar: TensorFlowFloatingPoint>: Layer {
     return (input - moments.mean) * inv + offset
   }
 }
+
+/// A layer that applies group normalization over a mini-batch of inputs.
+///
+/// Reference: [Group Normalization](https://arxiv.org/abs/1803.08494).
+@frozen
+public struct GroupNorm<Scalar: TensorFlowFloatingPoint>: Layer {
+  /// The offset value, also known as beta.
+  public var offset: Tensor<Scalar>
+  /// The scale value, also known as gamma.
+  public var scale: Tensor<Scalar>
+  /// The number of groups.
+  @noDerivative public let groups: Int
+  /// The axis.
+  @noDerivative public let axis: Int
+  /// The variance epsilon value.
+  @noDerivative public let epsilon: Scalar
+
+  /// Creates a group normalization layer.
+  public init(
+    offset: Tensor<Scalar>,
+    scale: Tensor<Scalar>,
+    groups: Int,
+    axis: Int,
+    epsilon: Scalar
+  ) {
+    precondition(axis != 0, "The axis cannot be batch axis.")
+    precondition(offset.rank == 1, "The offset must have rank 1.")
+    precondition(scale.rank == 1, "The scale must have rank 1.")
+    precondition(
+      offset.shape == scale.shape,
+      "The offset and the scale must have same shape.")
+    self.offset = offset
+    self.scale = scale
+    self.groups = groups
+    self.axis = axis
+    self.epsilon = epsilon
+  }
+
+  /// Creates a group normalization layer.
+  ///
+  /// - Parameters:
+  ///   - featureCount: The number of features.
+  ///   - groups: The number of groups.
+  ///   - axis: The axis that should be normalized.
+  ///   - epsilon: The small scalar added to variance.
+  public init(
+    featureCount: Int,
+    groups: Int,
+    axis: Int = -1,
+    epsilon: Scalar = 1e-3
+  ) {
+    precondition(
+      featureCount.isMultiple(of: groups),
+      "The feature count must be divisible by groups.")
+    self.init(
+      offset: Tensor(zeros: [featureCount]),
+      scale: Tensor(ones: [featureCount]),
+      groups: groups,
+      axis: axis,
+      epsilon: epsilon
+    )
+  }
+
+  /// Returns the output obtained from applying the layer to the given input.
```

Comment:
Nit: kinda wordy; how about ”Returns the result of applying this layer to `input`.”?

Reply:
There are many other `Returns the output obtained from applying the layer to the given input.` in this repository.
It seems this is the standard(?) comment for `Layer.callAsFunction`.
I leave it and wait someone will change all of them.